### PR TITLE
feat: Support creation of color maps through SLD filters

### PR DESCRIPTION
**Pull request by raitisbe**
## Description

Inside styler add a button to generate filters to categorize selected variable into 11 bins and set
a color for symbols from various colormaps supported by colormap library.

Can be seen in action on: https://mandolin.bosc.lv/ Track Šumavaprodukt - auto layer

image

BREAKING CHANGE: npm i colormaps
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [x] Dependency updates
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)
- [ ] I am unsure (we'll look into it together)

## Do you introduce a breaking change?

- [x] Yes
- [ ] No
- [ ] I am unsure (no worries, we'll find out)

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [MIT License]
- [x] I have followed the [guidelines for contributing](https://github.com/hslayers/hslayers-ng/blob/master/CONTRIBUTING.md)
- [x] The proposed change fits to the content of the [code of conduct](https://github.com/hslayers/hslayers-ng/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)
- [ ] I'm lost; why do I have to check so many boxes? Please help!

### CODE REVIEW

File: projects/hslayers/src/components/styles/styler.component.ts [hunk 1]
```diff
@@ -63,6 +64,7 @@ export class HsStylerComponent
       });
   }
   ngOnInit(): void {
+    debugger;
```
Comment (DailisLangovskis) [inline projects/hslayers/src/components/styles/styler.component.ts:67]: Probably forgot to remove

File: projects/hslayers/src/components/styles/styler.service.ts [hunk 2]
```diff
@@ -411,11 +412,64 @@ export class HsStylerService {
...
     const appRef = this.get(app);
     switch (kind) {
+      case 'ColorMap':
+        const colors = colormap({
+          colormap: colorMapName,
+          nshades: 11,
+          format: 'hex',
+          alpha: 1,
+        });
+        const attr = prompt(
```
Comment (DailisLangovskis) [inline projects/hslayers/src/components/styles/styler.service.ts:434]: Will need this as translation i suppose.

File: projects/hslayers/src/components/styles/styler.component.ts (hunk 1 above)
Comment (raitisbe) [reply projects/hslayers/src/components/styles/styler.component.ts:67]: Yes, indeed. Thanks

**Comment by DailisLangovskis** [comment]
@raitisbe There should be a check if provided attribute has valid number values, if not request the attribute again or cancel.

File: projects/hslayers/src/components/styles/styler.service.ts (hunk 2 above)
Comment (raitisbe) [reply projects/hslayers/src/components/styles/styler.service.ts:436]: Yes, added that

**Comment by raitisbe** [comment]
> @raitisbe There should be a check if provided attribute has valid number values, if not request the attribute again or cancel.

Maybe not, since the layer might not have any features at all at the time of the SLD creation

**Pull request merged**